service/index.js

```diff
@@ -240,6 +265,40 @@ module.exports = class AuthenticationService {
       if (!deletedCount && (strict === true)) return Promise.reject(new Error(`Failed to remove user: ${email}.`));
     });
   }
+
+  /**
+   * Authorizes third party then logs in or signs up the user.
+   */
+  authorizeThirdParty(code, provider) {
+    if (!code || !provider) {
+      return Promise.reject(
+        new Error('Required parameters missing for third party authorization.')
+      );
+    }
+
+    return p(this).deferrari.deferUntil(THIRD_PARTY_CONFIG)
+    .then(authieThirdParty => authieThirdParty.login(code, provider)
+      .then(result => p(this).deferrari.deferUntil(CONNECTED)
+        .then(models => models.AuthUser.findOne({where: {email: result.email}})
+          .then(authUser => {
+            return authUser ?
```

Comment:
can be shorted to:
`.then(authUser => authUser || models.AuthUser.create({email: result.email}))`